### ISSUE TITLE
Initialize picker to now() on init

### ISSUE THREAD

**Issue by joakimgk** [vlio20/angular-datepicker#332]
Adding the date picker to an Angular HTML template brings up the picker, and I bind the `ngModel` to a variable in the component. It's being shown as an (empty) input field/box. Clicking on it brings up the picker, and the chosen date value is set in the corresponding variable. At the same time, the chosen date is shown in the input box.

How do I set the *initial value* of the picker (to current time) when the page loads? I've tried setting the value of the bound variable, in the component's `ngOnInit`, but that is not reflected in the picker. It remains empty until you actually interact with the picker. I don't want to display a simple "placeholder" text.

**Comment by joakimgk** [comment]
`<dp-date-picker [(ngModel)]="pickerDate" mode="month" theme="dp-material" [config]="dateconfig" #dayPicker></dp-date-picker>`

**Comment by joakimgk** [comment]
```
  ngOnInit(): void {
     this.pickerDate = moment();  // something like this
  }
```

**Comment by joakimgk** [comment]
I tried making a method to set the date (moment) before opening the picker, but I always just get the current date when the picker is displayed

```
  test(): void {
    console.log("TEST set pickerDate");
    this.pickerDate = moment("20171031", "YYYYMMDD");
    if (this.datePicker) {
      this.datePicker.api.open();
    }
  }
```

**Comment by joakimgk** [comment]
See this issue is already raised

**Comment by vlio20** [comment]
Can you post a link to the already opened issue?

**Comment by joakimgk** [comment]
These two?
https://github.com/vlio20/angular-datepicker/issues/306
https://github.com/vlio20/angular-datepicker/issues/312

**Comment by vlio20** [comment]
Can you please post a stackblitz?

**Comment by vlio20** [comment]
I think this is a different bug than described in the mentioned links. Seems like the ngModel is ignored. Will mark this as a bug. Thanks @joakimgk

**Comment by joakimgk** [comment]
Excellent! Please let me know when there is some fix, so I can test this. You see, our app stores the user's last choice of date to local storage, so that state can be restored if the page is reloaded in the browser. Currently, this leads to the date picker and the chosen date to be out of sync.

**Comment by vlio20** [comment]
I see, I hope I will be able to put my hands on it in the next couple of days.

**Comment by vlio20** [comment]
@joakimgk, I just realized that you are using a `Date` object. You should only use string or moment to be passed to the ngModel. Here is a working example: https://stackblitz.com/edit/angular-rbmbzz

Let me know if I am missing something.

**Comment by joakimgk** [comment]
Hey

thanks, that helped! I got it working now. I made a fork of your Stackblitz
where the "do" button sets the date to Oct. 31 2017, rather than current
time.
The current date is set in the AppComponent's ngOnInit, on page load.
https://stackblitz.com/edit/angular-s7613r

However, there are still two other things which might still be a bug:
To reproduce:

1) loading the page sets current date
2) opening the picker, and current selection (jan 2018) is highlighted
3) choose another date: Mar 2018
4) re-opening the picker, and current selection (mar 2018) is highlighted
5) click "do" button, and current selection changes to "oct 2017"
6) re-opening the picker, but picker has now been re-initialized back to
start (jan 2018)!

**Comment by vlio20** [comment]
regarding point 6, if you change progrematically the date of the date-picker (point 5) the calendar would not change it displayed date. in order to move the calendar to the chosen date you would have to use another api - `moveCalendarTo` - this will be available in version 2.7.0. You can already find it in the documentation on the next branch.

Let me know if you have any other question. Also, it would be great if you could share a link/screenshot of your implementation of the date picker in your application.

Thanks!

**Comment by joakimgk** [comment]
Thanks, this works nicely now. I'll check back when the new version is out!

I've only got a working prototype, but I can send you some screenshots.


Joakim

10. jan. 2018 7.04 a.m. skrev "Vlad Ioffe" <notifications@github.com>:

regarding point 6, if you change progrematically the date of the
date-picker (point 5) the calendar would not change it displayed date. in
order to move the calendar to the chosen date you would have to use another
api - moveCalendarTo - this will be available in version 2.7.0. You can
already find it in the documentation on the next branch.

Let me know if you have any other question. Also, it would be great if you
could share a link/screenshot of your implementation of the date picker in
your application.

Thanks!

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
<https://github.com/vlio20/angular-datepicker/issues/332#issuecomment-356508542>,
or mute the thread

**Comment by vlio20** [comment]
Would be great
